packages/edit-post/src/components/manage-blocks-modal/manager.js

```diff
@@ -77,9 +78,14 @@ function BlockManager( {
 					<BlockManagerCategory
 						key={ category.slug }
 						category={ category }
-						blockTypes={ filter( blockTypes, {
-							category: category.slug,
-						} ) }
+						blockTypes={ filter(
+							blockTypes,
+							( blockType ) =>
+								blockType.category ===
+								( category === defaultCategory
+									? undefined
+									: category.slug )
+						) }
```

Comment:
Fun fact: This shorthand object predicate form for Lodash methods does not allow you to filter for `undefined` values, unless the property actually exists and is `undefined`. The function form works either way.

```js
const blockTypes = [ { name: 'example' } ];
lodash.filter( blockTypes, { category: undefined } );
// [] 😕

const blockTypes = [ { name: 'example', category: undefined } ];
lodash.filter( blockTypes, { category: undefined } );
// [ {...} ] 🤷‍♂️ (we can't rely on it)
```